client/src/main/java/io/confluent/kafka/schemaregistry/client/CachedSchemaRegistryClient.java

```diff
@@ -136,21 +135,19 @@ private int getIdFromRegistry(String subject, Schema schema)
   @Override
   public synchronized int register(String subject, Schema schema)
       throws IOException, RestClientException {
-    Map<Schema, Integer> schemaIdMap;
+    final IdentityLRUCache<Schema, Integer> schemaIdMap;
     if (schemaCache.containsKey(subject)) {
       schemaIdMap = schemaCache.get(subject);
     } else {
-      schemaIdMap = new IdentityHashMap<Schema, Integer>();
+      schemaIdMap = new IdentityLRUCache<>(identityMapCapacity);
       schemaCache.put(subject, schemaIdMap);
     }
 
-    if (schemaIdMap.containsKey(schema)) {
-      return schemaIdMap.get(schema);
+    Integer id = schemaIdMap.get(schema);
+    if (id != null) {
+      return id;
```

Comment:
It happens to forget a return from within a branch sometimes. How about: 
```java
if (id == null) {
...
}
return id;
```
instead?